app/src/main/java/com/android/sample/ui/listActivities/ViewListActivities.kt

```diff
@@ -176,12 +177,15 @@ fun ListActivitiesScreen(
               is ListActivitiesViewModel.ActivitiesUiState.Success -> {
                 var activitiesList =
                     (uiState as ListActivitiesViewModel.ActivitiesUiState.Success).activities
-                if (selectedIndex != 0) {
-                  activitiesList = activitiesList.filter { it.type.name == options[selectedIndex] }
+                if (checkedList.isNotEmpty()) {
+                  activitiesList =
+                      activitiesList.filter {
+                        checkedList.contains(categories.indexOf(it.category))
+                      }
                 }
                 activitiesList = activitiesList.filter { it.date >= Timestamp.now() }
                 if (activitiesList.isEmpty()) {
-                  if (selectedIndex == 0) {
+                  if (checkedList.isEmpty()) {
                     Text(
                         text = "There is no activity yet.",
```

Comment:
Is this the correct text to show in this case ? Also, text should be put in C file.